programming-extensions/programming-extension-dataspaces/src/main/java/org/objectweb/proactive/extensions/dataspaces/vfs/adapter/VFSFileObjectAdapter.java

```diff
@@ -601,29 +610,19 @@ private static boolean checkIfFileWritable(DataSpacesFileObject dsfo) throws Fil
         if (!dsfo.isWritable()) {
             return false;
         }
-        File file = null;
+
         try {
-            file = convertDataSpaceToFileIfPossible(dsfo);
+            File file = convertDataSpaceToFileIfPossible(dsfo);
+
+            if (file != null) {
+                return Files.isWritable(file.toPath());
+            }
         } catch (URISyntaxException e) {
             throw new FileSystemException(e);
         } catch (DataSpacesException e) {
             throw new FileSystemException(e);
         }
-        if (file != null) {
-            boolean answer = file.canWrite();
-            if (!answer) {
-                return false;
-            }
-            // in some cases the canWrite method is not accurate, we try to create a file to check real write permission
-            File tryFile = new File(file, "testWrite.ack");
```

Comment:
Probably the comment is not clear enough, and the code has issue related to file deletion, but as you can see on this thread:

http://stackoverflow.com/questions/9806303/file-canwrite-is-not-working-as-per-expectations

"I have also found that File.canWrite() cannot be trusted, especially over network drives, often returning true even though a file write will fail or vice-versa. I made my own method that actually tries to write a dummy file to the dir. That is simple to write and foolproof. Maybe they fixed it though."